src/cogs/snipe.py

```diff
@@ -25,11 +25,18 @@ def __init__(self, bot):
         self._message_history: dict[int, list[discord.Message]] = defaultdict(lambda: deque(maxlen=conf.max_edit_msg))
 
     @commands.hybrid_command(with_app_command=True)
-    async def snipe(self, ctx: Context, number: int = 0):
-        if ctx.channel.id not in self._deleted:
-            return await ctx.send("No message to snipe.")
+    async def snipe(self, ctx: Context, number: int = 0, channel: int = 0):
+        if channel == 0:
+            if ctx.channel.id not in self._deleted:
+                return await ctx.send("No message to snipe.")
+
+            deleted_msgs = self._deleted[ctx.channel.id]
+        else:
+            if channel not in self._deleted:
+                return await ctx.send("No message to snipe.")
+
+            deleted_msgs = self._deleted[channel]
```

Comment:
Please don't duplicate that much code, instead do something along the lines of `id ?: current channel id` and just keep the rest of it as is.

Reply:
that aint valid syntax but ok

Reply:
I'd assume kris knows the elvis operator considering how much C he programs (but yes, it's a gnu extension).